flocker/control/_persistence.py

```diff
@@ -87,7 +93,58 @@ def decode_object(dictionary):
                        [_DEPLOYMENT_FIELD], [])
 
 
-class ConfigurationPersistenceService(Service):
+class LeaseService(Service):
+    """
+    Manage leases.
+    In particular, clear out expired leases once a second.
+
+    :ivar _reactor: A ``twisted.internet.reactor`` implementation.
+    :ivar _persistence_service: The persistence service to act with.
+    :ivar _lc: A ``twisted.internet.task.LoopingCall`` run every second
+        to update the configured leases by releasing leases that have
+        expired.
+    """
+    def __init__(self, reactor, persistence_service):
+        if reactor is None:
+            reactor = default_reactor
+        self._reactor = reactor
+        self._persistence_service = persistence_service
+
+    def startService(self):
+        self._lc = LoopingCall(self._expire)
+        self._lc.clock = self._reactor
+        self._lc.start(1)
+
+    def stopService(self):
+        self._lc.stop()
+
+    def _expire(self):
+        now = datetime.now(tz=UTC)
+        return update_leases(lambda leases: leases.expire(now),
+                             self._persistence_service)
```

Comment:
So this is the only comment that remains (partially) unaddressed at this stage. Basically, this branch includes a minimal Lease service as part of the control service (now a `MultiService`) and this runs a loop every second to check for expired leases. However, in this branch, we aren't updating the `Deployment` model to be lease-aware (because that would break things if it were done now), which means our service here doesn't do anything yet. `_expire` here calls `update_leases`, which will save the configuration through the persistence service after running the `Leases` model's `expire` method. But in this branch it just returns a `Deferred` with a success result of `None`. The problem with doing anything further than I've done in this branch with this code and the skeletal test infrastructure in the persistence unit tests is that it's not really possible until `Deployment` has a concept of leases. Suggestions are welcome.